File: 08_May_2nd/gyeongwook/twoSum.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} target
    # @return {integer[]}
    def twoSum(self, nums, target):
        length = len(nums)
        lookup_table = {}
        
        for index2 in range(length) :
            if target-nums[index2] in lookup_table :
                return [lookup_table[target-nums[index2]]+1, index2+1]
            else :
                lookup_table[nums[index2]] = index2

File: 08_May_2nd/Seonghun/TwoSum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        if (nums == null || nums.length < 2)
            return null;
        
        int leng = nums.length;
        Map<Integer, Integer> numIdxMap = new HashMap<Integer, Integer>();
        
        for (int i = 0 ; i < leng ; i++) {
            
            int diff = target - nums[i];
            
            if (numIdxMap.containsKey(diff)) {
                
                int prvIdx = numIdxMap.get(diff);
                
                int[] result = new int[2];
                if (i < prvIdx) {
                    result[0] = i+1;
                    result[1] = prvIdx+1;
                } else {
                    result[0] = prvIdx+1;
                    result[1] = i+1;
                }
                
                return result;
            }
            
            numIdxMap.put(nums[i], i);
        }
        
        return null;
    }
}